softlayer/resource_softlayer_vlan.go

```diff
@@ -196,17 +201,27 @@ func resourceSoftLayerVlanRead(d *schema.ResourceData, meta interface{}) error {
 
 	// Subnets
 	subnets := make([]map[string]interface{}, 0)
+	primarySubnets := make([]map[string]interface{}, 0)
 
 	for _, elem := range vlan.Subnets {
 		subnet := make(map[string]interface{})
+		primarySubnet := make(map[string]interface{})
+		validPrimaryType := regexp.MustCompile(`.*PRIMARY.*`)
+		if validPrimaryType.MatchString(*elem.SubnetType) {
+			primarySubnet["subnet"] = fmt.Sprintf("%s/%s", *elem.NetworkIdentifier, strconv.Itoa(*elem.Cidr))
+			primarySubnet["subnet_type"] = *elem.SubnetType
+			primarySubnet["subnet_size"] = 1 << (uint)(32-*elem.Cidr)
+			primarySubnets = append(primarySubnets, primarySubnet)
+		}
 		subnet["subnet"] = fmt.Sprintf("%s/%s", *elem.NetworkIdentifier, strconv.Itoa(*elem.Cidr))
 		subnet["subnet_type"] = *elem.SubnetType
+		subnet["subnet_size"] = 1 << (uint)(32-*elem.Cidr)
 		subnets = append(subnets, subnet)
 	}
 	d.Set("subnets", subnets)
 
-	if vlan.Subnets != nil && len(vlan.Subnets) > 0 {
-		d.Set("subnet_size", 1<<(uint)(32-*vlan.Subnets[0].Cidr))
+	if primarySubnets != nil && len(primarySubnets) > 0 {
+		d.Set("subnet_size", primarySubnets[0]["subnet_size"])
```

Comment:
If there are no primarySubnets, should we fall back to the first element of `vlan.Subnets` for the subnet size?